node/node.go

```diff
@@ -658,6 +682,85 @@ func startStateSync(ssR *statesync.Reactor, bcR fastSyncReactor, conR *cs.Reacto
 	return nil
 }
 
+// NewSeedNode returns a new seed node, containing only p2p, pex reactor
+func NewSeedNode(config *cfg.Config,
+	nodeKey p2p.NodeKey,
+	genesisDocProvider GenesisDocProvider,
+	dbProvider DBProvider,
+	logger log.Logger,
+	options ...Option) (*Node, error) {
+
+	_, stateDB, err := initDBs(config, dbProvider)
+	if err != nil {
+		return nil, err
+	}
+
+	state, genDoc, err := LoadStateFromDBOrGenesisDocProvider(stateDB, genesisDocProvider)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
```suggestion
      genDoc, err := genesisDocProvider()
      if err != nil {
            return nil, err
      }
      
      state, err := sm.MakeGenesisState(genDoc)
      if err != nil {
               return nil, err
      }
```

IMO, I don't think it makes sense to have a db that just takes the state derived from genesis and is never updated so I would remove `dbProvider` and just have it like this